device/core/test/_client_test.js

```diff
@@ -942,157 +688,60 @@ describe('Client', function () {
-          */
-        process.nextTick(function() {
-          process.nextTick(function() {
-            callback(null, new results.Connected());
-          });
-        });
-      });
-
-      sinon.stub(dummyTransport, 'updateSharedAccessSignature', function(newSas, callback) {
-        callback(null, new results.SharedAccessSignatureUpdated(true));
-      });
-
+      var fakeError = new Error('fake');
+      sinon.stub(dummyTransport, 'enableC2D').callsFake(function (callback) { callback(fakeError); });
       var client = new Client(dummyTransport);
-      client.blobUploadClient = { updateSharedAccessSignature: function() {} };
-      client.open(function() {
-        client.updateSharedAccessSignature('newSas', function() {
-          // at that point the transport has reconnected, and the receiver should be reconnected too.
-          process.nextTick(function() {
-            receiver.emit('message', new Message());
-          });
-        });
+      client.on('error', function (err) {
+        assert.strictEqual(err, fakeError);
+        testCallback();
+      })
 
-        client.on('message', function () {
-          testCallback();
-        });
-      });
+      // Calling 'on' twice to make sure it's called only once on the receiver.
+      // It works because the test will fail if the test callback is called multiple times, and it's called for every time the 'message' event is subscribed on the receiver.
+      client.on('message', function () { });
+      assert.isTrue(dummyTransport.enableC2D.calledOnce);
     });
 
-    /*Tests_SRS_NODE_DEVICE_CLIENT_16_065: [The client shall connect the transport if needed to subscribe receive messages.]*/
-    it('receives messages if the event handler is registered while connecting', function(testCallback) {
-      var receiver = new EventEmitter();
+    it('emits an error if it fails to stop listening for messages', function (testCallback) {
       var dummyTransport = new FakeTransport();
-      sinon.stub(dummyTransport, 'getReceiver', function(callback) {
-        callback(null, receiver);
-      });
-      sinon.stub(dummyTransport, 'connect', function(callback) {
-        process.nextTick(function() {
-          process.nextTick(function() {
-            callback(null, new results.Connected());
-          });
-        });
-      });
-
+      var fakeError = new Error('fake');
+      sinon.spy(dummyTransport, 'enableC2D');
+      sinon.stub(dummyTransport, 'disableC2D').callsFake(function (callback) { callback(fakeError); });
       var client = new Client(dummyTransport);
-      client.open(function(){
-        receiver.emit('message', new Message());
-      });
-
-      client.on('message', function () {
+      client.on('error', function (err) {
+        assert.strictEqual(err, fakeError);
         testCallback();
-      });
+      })
+
+      // Calling 'on' twice to make sure it's called only once on the receiver.
```

Comment:
>'on' [](start = 17, length = 4)

This comment doesn't match the code.  Do you mean to call removeAllListeners twice?

Reply:
no, I guess it's just old copy/pasta. removing the comment.

---
In reply to: [145237862](https://github.com/Azure/azure-iot-sdk-node/pull/109#discussion_r145237862) [](ancestors = 145237862)